publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java

```diff
@@ -110,13 +117,25 @@ protected PublicationResponse processInput(CreatePublicationRequest input, Reque
         RightsRetentionsApplier.rrsApplierForNewPublication(newPublication, customer.getRightsRetentionStrategy(),
                                                             customerAwareUserContext.username()).handle();
 
-        var createdPublication = Resource.fromPublication(newPublication)
-                                     .persistNew(publicationService, customerAwareUserContext.userInstance());
-        setLocationHeader(createdPublication.getIdentifier());
+        var publicationResponse = attempt(() -> Resource.fromPublication(newPublication))
+            .map(resource -> resource.persistNew(publicationService, customerAwareUserContext.userInstance()))
+            .map(PublicationResponse::fromPublication)
+            .orElseThrow(this::mapException);
+        setLocationHeader(publicationResponse.getIdentifier());
 
-        return PublicationResponse.fromPublication(createdPublication);
+        return publicationResponse;
     }
 
+    private ApiGatewayException mapException(Failure<PublicationResponse> failure) {
+        return switch (failure.getException()) {
+            case NotFoundException ex -> ex;
+            case UnauthorizedException ex -> ex;
+            case MultipleCristinIdentifiersException ex -> new BadRequestException(MULTIPLE_CRISTIN_IDENTIFIERS_MESSAGE);
```

Comment:
Missing BadRequestException, possibly others we aren't testing. This causes `CreatePublicationHandlerTest.shouldReturnBadRequestWhenAnExternalClientTriesToCreatePublishedDocumentWithoutTitleAndDoiRef` to fail